src/mame/machine/agatkeyb.h

```diff
@@ -0,0 +1,82 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+#ifndef MAME_MACHINE_AGATKEYB_H
+#define MAME_MACHINE_AGATKEYB_H
+
+#pragma once
+
+#include "machine/keyboard.h"
+
+
+/***************************************************************************
+    DEVICE TYPE GLOBALS
+***************************************************************************/
+
+DECLARE_DEVICE_TYPE(AGAT_KEYBOARD, agat_keyboard_device)
+
+
+/***************************************************************************
+    REUSABLE I/O PORTS
+***************************************************************************/
+
+INPUT_PORTS_EXTERN( agat_keyboard );
```

Comment:
Is this key matrix actually likely to be used by things elsewhere, or would it be better to keep it confined to one compilation unit?

Reply:
I don't think any other drivers would use this keyboard...

Reply:
Better to not declare it `extern` – keep it contained and make it `static`.